test/rest/http.test.js

```diff
@@ -62,5 +62,26 @@ define(['ably', 'shared_helper', 'chai'], function (Ably, helper, chai) {
 
       done();
     });
+
+    it('Should handle no content responses', function (done) {
+      //Intercept Http.do with test
+
+      function testRequestHandler(_, __, ___, ____, _____, ______, callback) {
+        callback(null, null, null, false, 204);
+      }
+
+      rest.http.do = testRequestHandler;
+
+      let callbackCalled = false;
+      rest.request('GET', '/foo', {}, null, {}, (error, response) => {
+        expect(error).to.be.null;
+        expect(response.statusCode).to.equal(204);
+        expect(response.items).to.be.empty;
+        callbackCalled = true;
```

Comment:
i think it's nicer to just call `done` inside the callback here than to mutate a variable and test against it